test/reshape_elimination.cpp

```diff
@@ -106,3 +110,310 @@ TEST(reshape_elimination, dot_transpose_to_dot_w_transpose_args)
     ASSERT_EQ(gdot->get_argument(1)->get_argument(0), W);
     ASSERT_EQ(gdot->get_shape(), (Shape{1, 2}));
 }
+
+TEST(reshape_elimination, recurrent_reshapes)
```

Comment:
While there's nothing wrong with tests like this, I think it's best as a general rule to test that the _structure_ of the graph is actually modified as expected. As it stands, this test will pass even if the pass doesn't do anything. (Unless I am missing something.)

Reply:
That is true! My intention is to add a check to count the final number of Reshapes.